### DISCUSSION TITLE
Get protocol and host/domain name via react router

### DISCUSSION THREAD

**Discussion by Dmo16** [remix-run/react-router#8367] [Q&A]
Hello all,

Is it possible to get the protocol and domain name via react-router?

I've tried several things, from useLocation to useHistory, without any luck.

For reference, what I'm looking for is the equivalent of this: 

`window.location.hostname`

Or this: 

`window.location.protocol`

Note, currently using react-router v5

**Comment by MeiKatz** [comment]
I don't get why `window.location.hostname` should not work for react router ...

**Reply by Dmo16** [reply]
I'm not following.

Are you suggesting I use `window.location.hostname`, or are you saying there should be a way to do it via react-router, which is what I'm looking for?

**Reply by MeiKatz** [reply]
I suggest that you just use what JavaScript offers you as inbuilt tools. Neither the hostname nor the protocol can change without a full page refresh. Therefore just use the values from `window.location`.

**Reply by Dmo16** [reply]
I was specifically looking for a way to do it with react-router to keep all the code conventions the same, but if it can't do it then that's fine, I'll use the window api.

**Reply by kaushalkg1** [reply]
try - window.location.origin
